src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const ch of actions) {
+    const clonedState = { ...currentState };
+
+    if (ch.type === 'addProperties') {
+      for (const [key, value] of Object.entries(ch.extraData)) {
+        clonedState[key] = value;
+      }
+    }
+
+    if (ch.type === 'removeProperties') {
+      for (const key of ch.keysToRemove) {
+        delete clonedState[key];
+      }
+    }
+
+    if (ch.type === 'clear') {
+      for (const key in clonedState) {
+        delete clonedState[key];
+      }
+    }
+
+    result.push(clonedState);
+    currentState = clonedState;
```

Comment:
A default case should be added to the switch statement for error handling. This will catch any actions that don't match the expected types and can help to identify issues with the input data. [CODE STYLE]: switch/case should always have default case for error handling.